From f3454e10bd598aeee71692340f52009d03a73099 Mon Sep 17 00:00:00 2001
From: Luca Boccassi <luca.boccassi@microsoft.com>
Date: Wed, 13 May 2020 19:10:06 +0100
Subject: [PATCH 2/2] portablectl: use replace-unload when stopping a service
 with --no-reload

Once an image is detached, the service is no longer able to run.
Without a daemon-reload, if the unit is still loaded, a subsequent
attach will fail as the unit already exists.
Ensure it is always unloaded even without daemon-reload if --no-reload
is passed on the command line by using the new stop job mode.

(cherry picked from commit f15aa996977d3f52e0071a983456071de30d0018)
---
 src/portable/portablectl.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/portable/portablectl.c b/src/portable/portablectl.c
index fc83823cf8..0eec6d5a09 100644
--- a/src/portable/portablectl.c
+++ b/src/portable/portablectl.c
@@ -472,7 +472,7 @@ static int maybe_start_stop(sd_bus *bus, const char *path, bool start) {
                         start ? "StartUnit" : "StopUnit",
                         &error,
                         &reply,
-                        "ss", name, "replace");
+                        "ss", name, !start && !arg_reload ? "replace-unload" : "replace");
         if (r < 0)
                 return log_error_errno(r, "Failed to %s the portable service %s: %s",
                         start ? "start" : "stop", path, bus_error_message(&error, r));
-- 
2.20.1

